### PR TITLE
Test/e2e test guest mode sign in out

### LINKED ISSUE #440: Enhance current end 2 end tests
### *Story 1: Switch Accounts and Verify Profile Changes*  
The user signs into their first account, browses the app briefly, and then logs out. They sign into a second account and confirm that the profile and activity details now correspond to the new account, ensuring the correct profile information is displayed.

---

### *Story 2: Guest User Restrictions*  
A guest user navigates through the app, exploring activities, viewing maps, and attempting to interact with features like commenting or joining activities, only to see restricted access messages. Satisfied with the limitations, the guest returns to the sign-in page to create an account or log in.

### PR DESCRIPTION
**Pull request by hmaryam84**
## Summary

This pull request introduces 2 end 2 ends, hre are their description:

---

### **1. `aUserSwitchesProfile`**

**Purpose:** Test logging out of one account and logging into another.

1. **User 1 Login:**
   - Navigate to the Sign-In screen and enter User 1 credentials.
   - Verify navigation to the **Overview screen**.
   - Go to the **Profile** tab and confirm User 1's name is displayed.

2. **Log Out User 1:**
   - Use the **More Options** menu to log out.
   - Verify return to the **Sign-In screen**.

3. **User 2 Login:**
   - Enter User 2 credentials and log in.
   - Verify navigation to the **Overview screen**.
   - Go to the **Profile** tab and confirm User 2's name is displayed.

---

### **2. `guestCanSeeCorrectOverviewAndNavigateToActivityDetails`**

**Purpose:** Test guest navigation and feature restrictions.

1. **Guest Login:**
   - Click **Guest Sign-In** and verify navigation to the **Overview screen**.

2. **Filter and Search Activities:**
   - Apply category filters (e.g., **Sport**).
   - Use search functionality, including the `replace` feature, to refine results.

3. **View Activity Details:**
   - Click an activity card and verify key details are displayed.
   - Confirm guest restrictions (e.g., no **Enroll** button).

4. **Guest Profile and Liked Activities:**
   - Navigate to **Profile** and **Liked Activities** tabs to confirm prompts to sign in.

5. **Restart Guest Session:**
   - Return to **Sign-In** and re-enter as a guest. Verify navigation to **Overview**.

## Changes Made in the overall code

 **Refactoring:**
   - Updated `SignInViewModel` usage to replace direct calls to `Firebase.auth`, adhering to MVVM principles.
   - Applied `ktfmt` for consistent code formatting.
   
   closes #440

### CODE REVIEW

File: app/src/androidTest/java/com/android/sample/ui/endtoend/ActivityFlowTest.kt
Comment (mohamedtahaguelzim) [inline app/src/androidTest/java/com/android/sample/ui/endtoend/ActivityFlowTest.kt:127]: Thank you for adding these back !

File: app/src/androidTest/java/com/android/sample/ui/endtoend/ActivityFlowTest.kt
Comment (mohamedtahaguelzim) [inline app/src/androidTest/java/com/android/sample/ui/endtoend/ActivityFlowTest.kt:131]: Could have used the scroll argument.

File: app/src/androidTest/java/com/android/sample/ui/endtoend/ActivityFlowTest.kt
Comment (mohamedtahaguelzim) [inline app/src/androidTest/java/com/android/sample/ui/endtoend/ActivityFlowTest.kt:156]: Very meaningful ! Thanks !

File: app/src/androidTest/java/com/android/sample/ui/endtoend/ActivityFlowTest.kt
Comment (mohamedtahaguelzim) [inline app/src/androidTest/java/com/android/sample/ui/endtoend/ActivityFlowTest.kt:226]: Careful ! Here you use the same credentials, which is not what we are expected to do !! Will request changes !

File: app/src/androidTest/java/com/android/sample/ui/profile/ProfileScreenTests.kt
Comment (mohamedtahaguelzim) [inline app/src/androidTest/java/com/android/sample/ui/profile/ProfileScreenTests.kt:1]: Simple refactoring.

File: app/src/androidTest/java/com/android/sample/ui/endtoend/TestTags.kt
Comment (mohamedtahaguelzim) [inline app/src/androidTest/java/com/android/sample/ui/endtoend/TestTags.kt:1]: Thank you for adding these !

File: app/src/main/java/com/android/sample/resources/dummydata/E2E_DummyData.kt
Comment (mohamedtahaguelzim) [inline app/src/main/java/com/android/sample/resources/dummydata/E2E_DummyData.kt:121]: In that case I wouldn't name it defaultUser, but nvm it is not really important.

File: app/src/main/java/com/android/sample/ui/profile/Profile.kt
Comment (mohamedtahaguelzim) [inline app/src/main/java/com/android/sample/ui/profile/Profile.kt:448]: Excellent change !! This was not respecting MVVM model ! Thanks !

File: app/src/main/java/com/android/sample/ui/profile/Profile.kt
Comment (mohamedtahaguelzim) [inline app/src/main/java/com/android/sample/ui/profile/Profile.kt:452]: Well catched !

**Review by mohamedtahaguelzim — CHANGES_REQUESTED** [review]
Thank you so much for you help in the e2e tests. They are so much more meaningful now ! However, you did a small typo which cancels the purpose of the e2e. Should be pretty easy fix.

**Review by mohamedtahaguelzim — APPROVED** [review]
Thank you so much for such a quick response. Approved! Please wait for CI to pass before merging.

**Comment by hmaryam84** [comment]
> Thank you so much for you help in the e2e tests. They are so much more meaningful now ! However, you did a small typo which cancels the purpose of the e2e. Should be pretty easy fix.

Thank you so much for this warning, indeed i made a typo error, that made the test less meaninful. Icorrected it. Thank you again!

**Pull request merged**
